hatchet_sdk/clients/workflow_listener.py

```diff
@@ -249,6 +247,10 @@ async def _retry_subscribe(self):
                 if retries > 0:
                     await asyncio.sleep(DEFAULT_WORKFLOW_LISTENER_RETRY_INTERVAL)
 
+                # signal previous async iterator to stop
+                if self.curr_requester != 0:
+                    self.requests.put_nowait(self.curr_requester)
```

Comment:
Fixes some buggy behavior where the first parent workflow run to initiate a connection after a restart of the engine wasn't requesting workflow run ids on the correct listener channel. I think we were leaking `_request` methods and pushing into a queue which was shared by multiple request consumers/async iterators.